survey_sale_generation/models/survey_user_input.py

```diff
@@ -17,7 +17,7 @@ def _prepare_quotation(self):
             "partner_id": self.partner_id.id or self.create_uid.partner_id.id,
             "origin": self.survey_id.title,
             "survey_user_input_id": self.id,
-            "company_id": self.create_uid.company_id.id,
+            "company_id": self.env.company.id,
```

Comment:
Maybe you are generating the sale for one of your secondary companies. Isn't there any company data at survey or other level?

Reply:
Both the survey and the quote templates have the company_id field, but it may or may not be defined. The fact is that the budget must belong to a company, but setting the user's company is not correct, as the budget template and the survey may or may not belong to another company. For this reason, we have opted to use the company that is active at the time the budget is created.

Reply:
I know current code is incorrect, but the one you propose may not be correct either. According to what you say, this should use the survey's one, and fallback to this other. 

```suggestion
            "company_id": self.survey_id.company_id.id or self.env.company.id,
```

Reply:
But then you would have to add the dependency to survey_multi_company

Reply:
You haven't told me that! That's why I asked if there's the company field in any place. If there's no other company reference, then it's OK.

Reply:
That does not require dependencies, there is no other.